source/mainnet/smart-contracts/general/contract-instances.rst

```diff
@@ -152,6 +160,15 @@ reduce cost.
 
 .. _contract-instance-operations:
 
+Limits
+======
+
+When initializing or invoking a smart contract, the following limits are enforced:
+
+- The maximum parameter size is 65535 Bytes.
+- Up to 4 billion log items per execution.
+- The maximum return value size is 4 GigaBytes.
```

Comment:
I am not sure that we should mention these limits as you will hit the energy limit well before that. At least we should mention the energy limit.

Reply:
What is the current energy limit for the init/receive function?